package_tests/test_form.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+
+"""Test form generation."""
+
+from .base import init_package_base_structure
+from bobtemplates.plone import base
+from bobtemplates.plone import form
+from mrbob.configurator import Configurator
+from mrbob.configurator import Question
+
+import os
+
+
+def test_get_form_name():
+    question = Question(name="form_name", question="", default=None)
+    configurator = Configurator(
+        template="bobtemplates.plone:form",
+        target_directory="collective.foo.bar",
+        bobconfig={"non_interactive": True},
+        variables={"form_python_class_name": "FancyDemoForm"},
+    )
+    form.get_form_name_from_python_class(configurator, question)
+    assert question.default == "fancy-demo-form"
+
+
+def test_update_forms_configure_zcml(tmpdir):
+    """Test configure changes when changes are already in place."""
+    target_path = tmpdir.strpath + "/collective.sample"
+    package_path = target_path + "/src/collective/sample"
+    forms_path = package_path + "/forms/"
+    os.makedirs(target_path)
+    os.makedirs(package_path)
+    os.makedirs(forms_path)
+    template = """<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:plone="http://namespaces.plone.org/plone"
+    i18n_domain="{{{ package.dottedname }}}">
+
+  -*- extra stuff goes here -*-
+
+</configure>
+"""
+    with open(os.path.join(forms_path + "configure.zcml"), "w") as f:
+        f.write(template)
+    configurator = Configurator(
+        template="bobtemplates.plone:form",
+        target_directory="collective.sample",
+        bobconfig={"non_interactive": True},
+        variables={
+            "form_python_class_name": "MyForm",
```

Comment:
```suggestion
        target_directory=target_path,
        bobconfig={"non_interactive": True},
        variables={
            "form_python_file_name": "py_form",
            "form_python_class_name": "MyForm",
```

IMO `target_directory` needs to be an absolute path (or change current working directory to `tmpdir`), and `form_python_file_name` in configurator is required if not managed by `prepare_renderer`